freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/instance/DefaultInstanceGroupProvider.java

```diff
@@ -55,6 +56,11 @@ public Template createDefaultTemplate(CloudPlatform cloudPlatform, String accoun
                     entry(AwsInstanceTemplate.EBS_ENCRYPTION_ENABLED, Boolean.TRUE),
                     entry(InstanceTemplate.VOLUME_ENCRYPTION_KEY_TYPE, EncryptionType.DEFAULT.name()))));
         }
+        if (cloudPlatform == CloudPlatform.AZURE && entitlementService.isAzureDiskSSEWithCMKEnabled(accountId)) {
```

Comment:
Wrong condition. Quoting myself from https://github.com/hortonworks/cloudbreak/pull/10631#discussion_r656078740:

> Checking for the presence of the entitlement makes no sense; it is only a pre-requisite for using Disk SSE with CMK but does not imply that the feature has been actually requested during environment provisioning. The latter depends entirely on whether `diskEncryptionSetId` has been specified in the environment creation request or not.

In other words: replace the `entitlementService.isAzureDiskSSEWithCMKEnabled(accountId)` check with `diskEncryptionSetId != null`.